File: frontend/library-angular/src/app/book.ts
export interface Author {
  authorId: number;
  fullName: string;
}

export interface Category {
  categoryId: number;
  categoryName: string;
}

export interface Book {
  bookId: number;
  id?: number;
  title: string;
  authorId: number;
  authorEntity?: Author | null;
  author?: string;
  category?: string;
  categories: Category[];
}

File: backend/LibraryApi/LibraryApi/Models/Book.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace LibraryApi.Models
{
    public class Book
    {
        public int BookId { get; set; }

        [NotMapped]
        public int Id
        {
            get => BookId;
            set => BookId = value;
        }

        public string Title { get; set; } = string.Empty;

        public int AuthorId { get; set; }

        public Author? AuthorEntity { get; set; }

        [NotMapped]
        public string Author { get; set; } = string.Empty;

        [NotMapped]
        public string Category { get; set; } = string.Empty;

        public bool IsAvailable { get; set; } = true;

        public List<Category> Categories { get; set; } = new List<Category>();
    }
}